docs/source/conf.py

```diff
@@ -329,6 +336,56 @@
     ("py:class", "mlflow.models.signature.ModelSignature"),
 ]
 
+
+def _camel_to_snake(x):
+    return re.sub(r"(?<!^)(?=[A-Z])", "_", x).lower()
+
+
+def _get_reference_map():
+    """
+    Gets a dict that maps an invalid reference to a valid one.
+    """
+    ref_map = {
+        "mlflow.tracking.fluent.ActiveRun": "mlflow.ActiveRun",
```

Comment:
Sphinx tries to create a link for `ActiveRun` using its fully qualified class name (`mlflow.tracking.fluent.ActiveRun`), but our doc doesn't contain the reference `mlflow.tracking.fluent.ActiveRun` so it will result in the following warning:

```
/home/circleci/project/mlflow/__init__.py:docstring of mlflow.tracking.fluent.active_run:: WARNING: py:class reference target not found: mlflow.tracking.fluent.ActiveRun
```

As a workaround, map `mlflow.tracking.fluent.ActiveRun` to `mlflow.ActiveRun` (which exists).

Reply:
I am thinking for these:
```
mlflow.sklearn.xxx
mflow.pyspark.ml.xx
```
Will it prompot candidate list for possible "xxx" here ?
if no, can we add type hint for attributes like "mlflow.sklearn" ?

Reply:
Seems missing some similar cases, such as:

```
mlflow.deployments.BaseDeploymentClient
mflow.models.{Model, ModelSignature, ModelInputExample, infer_signature, FlavorBackend}
mflow.projects.SubmittedRun

mlflow.tracking.MlflowClient
mflow.types.{DataType, ColSpec, Schema, TensorSpec}
```

Reply:
Maybe we can automatically generate the `reference_map`, by searching all `__all__` exports in every sub-module, and for each item in `__all__` , if it is not a function but an Class, then we can add it into `reference_map`, we can generate the original ref by https://github.com/mlflow/mlflow/pull/4242/files#r614140548

Reply:
True, but we don't need to address them in this PR. I'll create a follow-up if necessary.

Reply:
One challenge is:

```
{"mlflow.tracking.fluent.ActiveRun": "???"}
```

How can we dynamically get `???` ?

Reply:
> How can we dynamically get < valid_ref > ?

Using module (include the __all__ list) name + item name?
we can search 2 depth level modules